arangod/Cluster/SynchronizeShard.cpp

```diff
@@ -236,12 +274,59 @@ static arangodb::Result addShardFollower(
     }
 
     uint64_t docCount;
-    {
+    if (recalc == ::ChecksumRecalculation::Self) {
+      // recalculate locally
+      LOG_TOPIC("29384", INFO, Logger::MAINTENANCE) 
+         << "recalculating collection count on follower for "
+         << database << "/" << shard;
+
+      Result res = collectionReCount(collection, docCount);
+      if (res.fail()) {
+        return res;
+      }
+    } else {
+      // use cached count
       Result res = collectionCount(collection, docCount);
       if (res.fail()) {
         return res;
       }
     }
+
+    if (recalc == ::ChecksumRecalculation::Leader) {
```

Comment:
The incremental sync actually does recalculate the counts on the master on the fly already.
Look at https://github.com/arangodb/arangodb/blob/devel/arangod/RocksDBEngine/RocksDBReplicationContext.cpp#L471